src/rust/engine/src/nodes/execute_process.rs

```diff
@@ -230,7 +230,9 @@ impl ExecuteProcess {
             .map_err(|e| throw(format!("Failed to serialize process: {e}")))?;
         workunit.update_metadata(|initial| {
             initial.map(|(initial, level)| {
-                let mut user_metadata = Vec::with_capacity(7);
+                let mut user_metadata = initial.user_metadata;
+                user_metadata.reserve(7);
```

Comment:
The new one in remote_cache executes first, so `initial.user_metadata` has two items in it here. Extend it instead of overwriting it